### PR TITLE
Adding All Files to New Branch

### PR DESCRIPTION
**Pull request by Kavish15**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "axios": "^0.27.2",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-redux": "^8.0.4",
+    "react-router-dom": "^6.4.1",
+    "react-scripts": "5.0.1",
+    "redux": "^4.2.0",
+    "web-vitals": "^2.1.4"
+  },
+  "scripts": {
```
Comment (jesusverma) [inline package.json:26]: dev dependecies

File: package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
```
Comment (jesusverma) [inline package.json:31]: node version > 14.0

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
```
Comment (jesusverma) [inline public/index.html:10]: meta use for SEO purpose
use some meta tag for SEO and change content

File: public/manifest.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (jesusverma) [inline public/manifest.json]: change a/t vatika

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Footer from './components/Common/Footer';
+import Navbar from './components/Common/Navbar';
+import Path from './Paths/Path';
+
+function App() {
+  return (
+    <>
+      <Navbar/>
+      <Path/>
```
Comment (jesusverma) [inline src/App.js]: path-> route.. path file should at root of src

File: src/Paths/Path.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { Route, Routes } from "react-router-dom";
+import LandingPage from "../components/Landing/subSection/LandingPage";
+import Product from "../components/Product/Product";
+import Blog from "../components/Blog/Blog";
+import AboutUs from "../components/About Us/AboutUs";
+import Plant from "../components/ProductDetails/Plant";
+import Cart from "../components/Cart/Cart";
+import Confirmation from "../components/Confirmation/Confirmation";
+import Checkout from "../components/Checkout/Checkout";
```
Comment (jesusverma) [inline src/Paths/Path.js]: lazzy loading

Comment (jesusverma) [reply src/Paths/Path.js]: code splitting

File: src/Paths/Path.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        <Route path="/" element={<LandingPage />} />
+        <Route path="/blog" element={<Blog />} />
+        <Route path="/aboutus" element={<AboutUs />} />
+        <Route path="/product" element={<Product />} />
+        <Route path="/product/:name" element={<Plant />} />
+        <Route path="/cart" element={<Cart />} />
+        <Route path="/confirmation" element={<Confirmation />} />
+        <Route path="/payment" element={<Checkout />} />
+        <Route>404 Page Not Found!</Route>
+      </Routes>
```
Comment (jesusverma) [inline src/Paths/Path.js]: create private route for the modules where token is mandatory.

File: src/components/About Us/AboutUs.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+// export default AboutUs;
+
+import React from "react";
+import "./AboutUs.css";
+import aboutus1 from "../../assets/aboutus1.png";
+import aboutus2 from "../../assets/aboutus2.png";
+import aboutus3 from "../../assets/aboutus3.png";
+
+function AboutUs() {
+  return (
```
Comment (jesusverma) [inline src/components/About Us/AboutUs.js:5]: memoization

File: src/App.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Footer from './components/Common/Footer';
+import Navbar from './components/Common/Navbar';
+import Path from './Paths/Path';
+
+function App() {
+  return (
+    <>
```
Comment (jesusverma) [inline src/App.js:10]: Error boundary

File: src/components/Blog/Blog.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import Heading from "../Common/Heading";
+import "./Blog.css";
+
+function Blog() {
+  return (
```
Comment (jesusverma) [inline src/components/Blog/Blog.js:7]: memoization

File: src/components/Cart/Cart.js [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+  // console.log(grandTotal)
+  const dispatch = useDispatch();
+  const [total, setTotal] = useState(0);
+
+  useEffect(() => {
+    setTotal(
+      cart.reduce((acc, curr) => acc + Number(curr.price) * curr.qty, 0)
+    );
+    dispatch(cartTotal({ total: total }));
+  }, [cart, total]);
```
Comment (jesusverma) [inline src/components/Cart/Cart.js]: wrape total logic in a func

File: src/components/Cart/Cart.js [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+  const deleteItem = () => {
+    notification.error({
+      message: "Item Deleted",
+      placement: "top",
+      duration: 1.5,
+    });
+  };
+
+  return (
+    <>
```
Comment (jesusverma) [inline src/components/Cart/Cart.js:37]: split compoenent,. and any com ponet shjould cross more than 50line

File: src/components/Cart/Cart.js [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+              <h2>
+                Total
+                <span>
+                  Rs {cart.length === 0 ? "0" : grandTotal.total + 30}
+                </span>
+              </h2>
+            </div>
+            {cart.length === 0 ? (
+              <button className="shopping">Continue Shopping</button>
+            ) : (
```
Comment (jesusverma) [inline src/components/Cart/Cart.js]: ternary operatoor , and operator. use component

File: src/components/Checkout/Checkout.js [hunk 14]
```diff
@@ -0,0 +1,487 @@
...
+import { setAddress } from "../../redux/actions/productActions";
+
+const { Step } = Steps;
+
+const Checkout = () => {
+  const [form] = Form.useForm();
+  const [oneAddress, setOneAddress] = useState({});
+  const [toggleButton, setToggleButton] = useState(false);
+  const [coin, setCoin] = useState(0);
+
```
Comment (jesusverma) [inline src/components/Checkout/Checkout.js:26]: 1. const  [state,setState] = useState({coin:0,toggleButton:false});
2. if you have more than 3 state : useReducer

File: src/components/Checkout/Checkout.js [hunk 15]
```diff
@@ -0,0 +1,487 @@
...
+  function selectClick() {
+    setToggleButton((toggleButton) => !toggleButton);
+  }
+
+  const toggleClass = toggleButton ? "selected" : "";
+  const newCardClass = toggleButton ? "newCard" : "";
+
+  const addedAddress = (data) => {
+    if (data === "Address Already Saved") {
```
Comment (jesusverma) [inline src/components/Checkout/Checkout.js]: status code , succees: true/false

File: src/components/Checkout/Checkout.js [hunk 16]
```diff
@@ -0,0 +1,487 @@
...
+        message: `${data}`,
+        placement: "bottom",
+        duration: 1.5,
+      });
+      form.resetFields();
+    }
+  };
+
+  const moreThenThreeAddress = () => {
+    notification.error({
```
Comment (jesusverma) [inline src/components/Checkout/Checkout.js]: error , success component

File: src/components/Checkout/Checkout.js [hunk 17]
```diff
@@ -0,0 +1,487 @@
...
+      message:"List Full , Delete Few Address !",
+      placement:'bottom',
+      duration:1.5,
+    });
+  };
+
+  const onAddressSaveAPI = async (values) => {
+    const value = { ...values, email: "kavishgarg15@gmail.com" };
+    const response = await axios
```
Comment (jesusverma) [inline src/components/Checkout/Checkout.js:33]: create  axios wrapper
https://gist.github.com/trkrameshkumar/804a7abd63622e715d2b5ee3afc5c249

File: src/components/Checkout/Checkout.js [hunk 18]
```diff
@@ -0,0 +1,487 @@
...
+      .post("http://localhost:5000/address", value)
```
Comment (jesusverma) [inline src/components/Checkout/Checkout.js]: url should in env file.

Comment (jesusverma) [reply src/components/Checkout/Checkout.js]: react scripts public env file

File: src/components/Checkout/Checkout.js [hunk 19]
```diff
@@ -0,0 +1,487 @@
...
+      });
+      console.log(response.data);
+      initPayment(response.data.data);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const initPayment = (data) => {
+    const options = {
```
Comment (jesusverma) [inline src/components/Checkout/Checkout.js]: use function here : createPayload()

File: src/components/Checkout/Checkout.js [hunk 20]
```diff
@@ -0,0 +1,487 @@
...
+                  } else {
+                    onAddressSaveAPI(values);
+                  }
+                }}
+                form={form}
+              >
+                <div className="fullName">
+                  <Form.Item
+                    name="firstName"
+                    rules={[
```
Comment (jesusverma) [inline src/components/Checkout/Checkout.js]: const rules = {[
                      {
                        required: true,
                        message: "Please enter First Name",
                      },
                      {
                        whitespace: true,
                        message: "First Name cannot be Empty",
                      },
                      {
                        min: 3,
                        message: "First Name must be Minimum 3 characters",
                      },
                      {
                        max: 15,
                        message: "First Name must be Under 15 characters",
                      },
                      {
                        type: Number,
                        message: "Number Not Allowed in First Name",
                      },
                    ]}
                    
                    
                    
                    
                    //////
                                      <Form.Item  rules=rules

File: src/components/Common/Footer.js [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+          <p>FAQ</p>
+        </div>
+
+        <div className="section-3"></div>
+
+        <div className="section-4">
+          <h1>Follow Us</h1>
+          <img src={require("../../assets/facebook.png")} alt="facebook" />
+          <img src={require("../../assets/youtube.png")} alt="youtube" />
+          <img src={require("../../assets/twitter.png")} alt="twitter" />
```
Comment (jesusverma) [inline src/components/Common/Footer.js]: always import asset files

File: src/components/Landing/subSection/Contact.js [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+
+function Contact() {
+  const saveMessage = () => {
+    notification.success({
+      message: "Sent, Team will Connect 🔔",
+      placement: "top",
+      duration: 2,
+    });
+    var form = document.getElementById("useForm");
+    form.reset();
```
Comment (jesusverma) [inline src/components/Landing/subSection/Contact.js:15]: formik

Comment (jesusverma) [reply src/components/Landing/subSection/Contact.js:15]: antd form

File: src/components/Product/Card.js [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+
+  const filter = useSelector((state) => state.filter);
+
+  const transformProduct = () => {
+    let sortedProduct = products;
+
+    if (filter.sort) {
+      sortedProduct = sortedProduct.sort((a, b) =>
+        filter.sort === "lowToHigh" ? a.price - b.price : b.price - a.price
+      );
```
Comment (jesusverma) [inline src/components/Product/Card.js]: return

File: src/components/Loading/Loading.js [hunk 24]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import "./Loading.css";
+
+function Loading() {
+  return (
+    <div className="loading">
+      <p>Loading....</p>
+    </div>
```
Comment (jesusverma) [inline src/components/Loading/Loading.js:23]: antd loader

File: package.json (hunk 1 above)
Comment (Kavish15) [reply package.json:26]: For Now, I Added Just Prettier, I will Add Jest and more later.

File: package.json (hunk 2 above)
Comment (Kavish15) [reply package.json:31]: Done

File: public/index.html (hunk 3 above)
Comment (Kavish15) [reply public/index.html:10]: Done

File: public/manifest.json (hunk 4 above)
Comment (Kavish15) [reply public/manifest.json]: Removed ManiFest.json

File: src/App.js (hunk 5 above)
Comment (Kavish15) [reply src/App.js]: Renamed Path to Route and added in Src root

File: src/Paths/Path.js (hunk 7 above)
Comment (Kavish15) [reply src/Paths/Path.js]: Done

File: src/Paths/Path.js (hunk 6 above)
Comment (Kavish15) [reply src/Paths/Path.js]: Done

File: src/components/About Us/AboutUs.js (hunk 8 above)
Comment (Kavish15) [reply src/components/About Us/AboutUs.js:5]: Actually i am not getting why to use Memoization here for About Us Page

File: src/components/Blog/Blog.js (hunk 10 above)
Comment (Kavish15) [reply src/components/Blog/Blog.js:7]: Actually, I am not getting why to use Memoization here for the Blog Page

File: src/App.js (hunk 9 above)
Comment (Kavish15) [reply src/App.js:10]: Should we add a react-error-boundary package?

File: src/components/Cart/Cart.js (hunk 11 above)
Comment (Kavish15) [reply src/components/Cart/Cart.js]: Done

File: src/components/Checkout/Checkout.js (hunk 18 above)
Comment (Kavish15) [reply src/components/Checkout/Checkout.js]: Done

File: src/components/Landing/subSection/Contact.js (hunk 22 above)
Comment (Kavish15) [reply src/components/Landing/subSection/Contact.js:15]: Done

File: src/components/Product/Card.js (hunk 23 above)
Comment (Kavish15) [reply src/components/Product/Card.js]: Done

File: src/components/Checkout/Checkout.js (hunk 20 above)
Comment (Kavish15) [reply src/components/Checkout/Checkout.js]: Done

File: src/components/Common/Footer.js (hunk 21 above)
Comment (Kavish15) [reply src/components/Common/Footer.js]: Will Be Moving All Asset Picture to Cloudinary

File: src/components/Checkout/Checkout.js (hunk 19 above)
Comment (Kavish15) [reply src/components/Checkout/Checkout.js]: Done

File: src/components/Checkout/Checkout.js (hunk 16 above)
Comment (Kavish15) [reply src/components/Checkout/Checkout.js]: Done

File: src/components/Checkout/Checkout.js (hunk 14 above)
Comment (Kavish15) [reply src/components/Checkout/Checkout.js:26]: Done

File: src/components/Cart/Cart.js (hunk 13 above)
Comment (Kavish15) [reply src/components/Cart/Cart.js]: Done

File: src/components/Cart/Cart.js (hunk 12 above)
Comment (Kavish15) [reply src/components/Cart/Cart.js:37]: Done

File: src/components/Loading/Loading.js (hunk 24 above)
Comment (Kavish15) [reply src/components/Loading/Loading.js:23]: Done

File: src/components/Checkout/Checkout.js (hunk 15 above)
Comment (Kavish15) [reply src/components/Checkout/Checkout.js]: Done

File: src/components/Blog/Blog.js (hunk 10 above)
Comment (Kavish15) [reply src/components/Blog/Blog.js:7]: Done

File: src/App.js (hunk 9 above)
Comment (Kavish15) [reply src/App.js:10]: Done

File: src/components/About Us/AboutUs.js (hunk 8 above)
Comment (Kavish15) [reply src/components/About Us/AboutUs.js:5]: Solved

**Comment by jesusverma** [comment]
LGTM

**Pull request merged**